Server/Controllers/PostsController.cs

```diff
@@ -0,0 +1,204 @@
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        // GET api/<PostsController>/5
+        [HttpGet("{id}")]
+        public IActionResult GetPostsById(int id)
+        {
+            try
+            {
+                using (fbContext fb = new fbContext())
+                {
+
+                    return Ok(fb.Posts.First(x => x.PostId == id));
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        // POST api/<PostsController>
+        [HttpPost]
+        public IActionResult AddNewPost([FromBody] Post post)
+        {
+            try
+            {
+                using (fbContext fb = new fbContext())
+                {
+                    post.CreatedDate = DateTime.Now;
+                    post.UpdatedDate = null;
+                    post.Likes = 0;
+                    post.Hearts = 0;
+                    fb.Posts.Add(post);
+                    fb.SaveChanges();
+                    return Ok(post);
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+
+        // PUT api/<PostsController>/5
+        [HttpPut("{id}")]
+        public IActionResult ModifyExistingPost(int id, [FromBody] Post post)
+        {
+            try
+            {
+                using (fbContext fb = new fbContext())
+                {
+                    var update = fb.Posts.First(x => x.PostId == id);
+                    update.PostName = post.PostName;
+                    update.PostDescription = post.PostDescription;
+                    update.UpdatedDate = DateTime.Now;
+                    fb.SaveChanges();
+                    return Ok(fb.Posts.First(x => x.PostId == id));
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+        [HttpPut("likes/{id}")]
+        public IActionResult AddLike(int id, [FromBody] Post post)
+        {
+            try
+            {
+                using (fbContext fb = new fbContext())
+                {
+                    var update = fb.Posts.First(x => x.PostId == id);
+                    update.Likes = post.Likes + 1;
+                    fb.SaveChanges();
+                    return Ok(update.Likes);
+                }
+            }
+            catch (Exception)
+            {
+                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            }
+        }
+        [HttpPut("dislikes/{id}")]
+        public IActionResult UnLike(int id, [FromBody] Post post)
```

Comment:
this should be present and one api endpoint will serve it purpose to update like or heart counter, so remove this

Reply:
Removed the method